src/backend/mod.rs

```diff
@@ -0,0 +1,8 @@
+#[cfg(any(feature = "s3-async-native", feature = "s3-async-rustls"))]
+pub(crate) mod s3;
+
+#[cfg(feature = "http-async")]
+pub(crate) mod http;
+
+#[cfg(feature = "mmap-async-tokio")]
+pub(crate) mod mmap;
```

Comment:
you don't really need to control the access here - enough to say `pub mod s3` - that shouldn't make it public on the crate level, as long as you do not use `pub mod backend` in `lib.rs`.

That said, mod.rs offer a better way to compartmentalize things - you don't need to expose internals of the `backend` mod to the rest of the crate - I think it is cleaner to expose just the needed structs (up to you though) - you can also just remove the `(crate)` and keep sub-namespaces.

```suggestion
#[cfg(any(feature = "s3-async-native", feature = "s3-async-rustls"))]
mod s3;

#[cfg(any(feature = "s3-async-native", feature = "s3-async-rustls"))]
pub use s3::S3Backend;

#[cfg(feature = "http-async")]
mod http;

#[cfg(feature = "http-async")]
pub use http::HttpBackend;

#[cfg(feature = "mmap-async-tokio")]
mod mmap;

#[cfg(feature = "mmap-async-tokio")]
pub use mmap::MmapBackend;
```

Reply:
Nice catch, cleaned up.